38.md

```diff
@@ -0,0 +1,170 @@
+    ...
+}
+```
+
+The members of the group ( people who are allowed to post there) should be added as p tags to this event. The creator should also add themselves as p tag, if they want to participate there too.
+
+On creating a channel, the creator should also generate a unique/new private-public key pair which will serve as the `shared-secret` for a given channel. The creator of the channel should then communicate this secret to other members as described in section for `kind 104`.
+
+Note: Clients for a user may create a channel where the only participant is the creator of the channel. Then new members can be added by sending a kind 141 message, as discussed further. 
+
+## Kind 141: Update metadata and Add participants
+
+Update an encrypted channel's public metadata including its particpants.
+
+Clients and relays SHOULD handle kind 141 events similar to kind 0 `metadata` events.
+
+Clients SHOULD ignore kind 141s from pubkeys other than the pubkey that created the given channel using the corresponding kind 140 event. In other words, only the creator of channel can send kind 141 events for a given channel; kind 141 for any channel sent by other people, even participants of that channel, are ignored.
+
+Clients SHOULD support basic metadata fields:
+
+- `name` - string - Channel name
+- `about` - string - Channel description
+- `picture` - string - URL of channel picture
+
+Clients MAY add additional metadata fields.
+
+Clients SHOULD use [NIP-10](10.md) marked "e" tags to recommend a relay.
+
+```json
+{
+    "content": "{\"name\": \"Updated Demo Channel\", \"about\": \"Updating a test channel.\", \"picture\": \"https://placekitten.com/201/201\"}",
+    "tags": [["e", <channel_create_event_id> <relay-url>]],
+    ...
+}
+```
+
+Further, the main use case of kind 141 event is to update the participants list. New p tags may be added, and old ones removed with the 141 event for an encrypted channel. Client will internally then manage participants. It should be noted that since the shared-secret is not changeable,  the 'removed' members can continue to read all channel messages.
+
+If new members are added, then its the responsibility of the creator's client to share the shared-secret with them as mentioned in kind 140 section. 
+
+## Kind 142: Send encrypted message to encrypted group channel
+
+
+Clients SHOULD use [NIP-10](10.md) marked "e" tags to recommend a relay and specify whether it is a reply or root message.
+
+Clients SHOULD append [NIP-10](10.md) "p" tags to replies.
+
+### Format
+
+Root message:
+
+```json
+{
+    "content": <string>,
+    "tags": [["e", <kind_140_event_id>],
+    ...
+}
+```
+
+Reply to another message:
+
+```json
+{
+    "content": <string>,
+    "tags": [
+        ["e", <kind_140_event_id>],
+        ["e", <kind_142_event_id_replied_to>], 
+        ["p", <pubkey> <relay-url>],
+        
+        ...
+    ],
+    ...
+}
+```
+
+## Kind 104: Communicate the Shared-Secret
```

Comment:
This part could handle encryption the way it is done on https://github.com/nostr-protocol/nips/pull/168 instead. This creates a single DM that can be read by multiple people by encrypting it with a unique secret that is included in the tags for every recipient.

This avoids the channel owner needing to send N messages every time someone is removed from the group.